### ISSUE TITLE
Intermittent Audio

### ISSUE THREAD

**Issue by s1rc** [KhaosT/homebridge-camera-ffmpeg#286]
I’ve got the video working for my two UniFi G3-Flex test cameras pretty easily. 

Audio works after recompiling ffmpeg as per the wiki (running in a jail on my FreeNAS server). However the audio cuts in and out every second, I’m not sure what the issue could be. The rtsp streams works with sound in VLC, so I can assume that’s not the issue. 

Here are my camera configs:
```
       {
            "platform": "Camera-ffmpeg",
            "cameras": [
                {
                    "name": "Front Door",
                    "videoConfig": {
                        "source": "-rtsp_transport http -re -i rtsp://x.x.x.x:7447/xxxxxx",
                        "audio": true,
                        "packetsize": 188,
                        "maxStreams": 2,
                        "maxWidth": 1920,
                        "maxHeight": 1080,
                        "maxFPS": 25,
                        "mapvideo": "0:1",
                        "mapaudio": "0:0",
                        "debug": true
                    }
                },
                {
                    "name": "Garage",
                    "videoConfig": {
                        "source": "-rtsp_transport http -re -i rtsp://x.x.x.x:7447/xxxxxx",
                        "audio": true,
                        "packetsize": 188,
                        "maxStreams": 2,
                        "maxWidth": 1920,
                        "maxHeight": 1080,
                        "maxFPS": 25,
                        "mapvideo": "0:1",
                        "mapaudio": "0:0",
                        "debug": true
                    }
                }
            ]
        }
```

**Comment by s1rc** [comment]
Still having this issue. I've been trying different settings but I can't seem to get consistent audio. I've attached a screen capture of the audio issue, as well as a log excerpt.

[Screen Capture](https://youtu.be/tLOPrvSnA3k)

[homebridge-out.log](https://github.com/KhaosT/homebridge-camera-ffmpeg/files/3127872/homebridge-out.log)

**Comment by fearlesschicken** [comment]
You have to play around with the packetSize setting:
packetSize If audio or video is choppy try a smaller value, set to a multiple of 188, default 1316

On mine, I stepped down from 1316 and kept testing until I found the value that worked.

**Comment by link266** [comment]
I am experiencing the exact same choppy audio issues as well. :(

I've tried every single option between 188 - 1316 (with multiples of 188 of course, so 188, 376, 564, etc) and every single one of them still resulted in the basically un-listenable choppy audio.

@fearlesschicken, are you able to post the settings from your JSON to compare?  Perhaps some audio options in the "additionalCommandline" field would help..??

**Comment by fearlesschicken** [comment]
Nothing special in my config...

`                                                       "videoConfig": {
                                                                                "source": "-rtsp_transport tcp -re -i rtsp://192.168.63.100:554/s0",
                                                                                "stillImageSource": "-rtsp_transport tcp -re -i rtsp://192.168.63.100:554/s1",
                                                                                "maxStreams": 3,
                                                                                "maxWidth": 1920,
                                                                                "maxHeight": 1080,
                                                                                "maxBitrate": 6000,
                                                                                "maxFPS": 30,
                                                                                "mapvideo": "0:1",
                                                                                "mapaudio": "0:0",
                                                                                "audio": true,
                                                                                "packetSize": 376
                                                                        }

                                                },
                                                {
                                                        "name": "Front Door",
                                                        "videoConfig": {
                                                                                "source": "-rtsp_transport tcp -re -i rtsp://192.168.63.101:554/s0",
                                                                                "stillImageSource": "-rtsp_transport tcp -re -i rtsp://192.168.63.101:554/s1",
                                                                                "maxStreams": 3,
                                                                                "maxWidth": 1920,
                                                                                "maxHeight": 1080,
                                                                                "maxBitrate": 6000,
                                                                                "maxFPS": 25,
                                                                                "mapvideo": "0:1",
                                                                                "mapaudio": "0:0",
                                                                                "audio": true,
                                                                                "packetSize": 376
                                                                        }

                                                },
`

Two unifi cameras - a g3 dome and and g3 flex

**Comment by link266** [comment]
Ok, thanks.  I'll experiment a little more and report back if I am able to get the audio working.

One question @fearlesschicken... Any specific reason why you use "tcp" (instead of "http") in your source?  Like, why use this: 

"source": "-rtsp_transport tcp -re -i rtsp://192.168.63.101:554/s0",

Instead of: 

"source": "-rtsp_transport http -re -i rtsp://192.168.63.101:554/s0",

?

In the guide I found to originally set all this up on the UniFi forums (link below) they specifically used "http" in the JSON so that's what I used but I noticed that yours is different and was curious if there was a reason.  

https://community.ui.com/questions/UniFi-Protect-with-HomeKit-Setup-Guide/3c47cc46-560c-4fcc-9eaa-d159a16c77e0?page=1
